R/helpers.R

```diff
@@ -454,3 +566,81 @@ checkDoubleArgumentsDesignObject <- function(designObj, ...) {
 
   }
 }
+
+# ---------- Boot helpers --------
+
+#' Computes the bootObj for sequential sampling procedures regarding nPlan, beta, the implied target
+#'
+#' @inheritParams designSafeZ
+#' @param values numeric vector. If objType equals "nPlan" or "beta" then values should be stopping times,
+#' if objType equals "logImpliedTarget" then values should be eValues.
+#' @param nBoot integer > 0 representing the number of bootstrap samples to assess the accuracy of
+#' approximation of the power, the planned sample size(s) of the safe test under continuous monitoring.
+#' @param nPlan integer > 0 representing the number of planned samples (for the first group).
+#' @param objType character string either "nPlan", "beta", or "logImpliedTarget".
+#'
+#' @return bootObj
+#' @export
+#'
+#' @examples
+#' computeBootObj(1:100, objType="nPlan", beta=0.3)
+computeBootObj <- function(values, beta=NULL, nPlan=NULL, nBoot=1e3L,
+                           objType=c("nPlan", "beta", "logImpliedTarget")) {
+  objType <- match.arg(objType)
+
+  if (objType=="beta") {
+    if (is.null(nPlan) || nPlan <= 0) stop("Please provide an nPlan > 0")
+
+    times <- values
+    stopifnot(nPlan > 0)
+
+    bootObj <- boot::boot(times,
+                          function(x, idx) {
+                            1-mean(x[idx] <= nPlan)
+                          },  R = nBoot)
+  } else if (objType=="nPlan") {
+    if (is.null(beta) || beta <= 0 || beta >= 1) stop("Please provide a beta in (0, 1)")
+
+    times <- values
+    bootObj <- boot::boot(times,
+                          function(x, idx) {
+                            quantile(x[idx], prob=1-beta, names=FALSE)
```

Comment:
Do I understand correctly that for bootstrapping nPlan, no values are set to Inf?

Reply:
Yes that is true, I want to avoid having nPlan being Inf. The trick that I use is to first compute nPlanBatch. This is my upperbound for nPlan. So in the worst case scenario (due to poor sampling) the optional stopping nPlan would equal nPlanBatch.